src/main/java/com/czertainly/core/service/impl/CertificateServiceImpl.java

```diff
@@ -1745,6 +1798,10 @@ private String downloadChain(String chainUrl) {
 
     public void switchRaProfile(SecuredUUID uuid, SecuredUUID raProfileUuid) throws NotFoundException, CertificateOperationException, AttributeException {
         Certificate certificate = getCertificateEntity(uuid);
+        if (certificate.isArchived()) {
+            logger.info("Certificate with UUID {} is archived and its RA Profile will not be updated.", uuid);
```

Comment:
Here is more suitable to throw Exception since it is user triggered operation that is not allowed. It should say it is not allowed to change RA profile for archived certificate.